tests/containers/podman_docker_interop.pm

```diff
@@ -0,0 +1,126 @@
+    my $docker = $self->containers_factory('docker');
+    $docker->cleanup_system_host();
+
+    # Stop docker daemon and start firewalld
+    systemctl("disable --now docker");
+    systemctl("enable --now firewalld");
+
+    systemctl("is-active docker", expect_false => 1);
+    systemctl("is-active firewalld");
+
+    record_info('Firewalld Zones', script_output("firewall-cmd --zone=public --list-all && firewall-cmd --get-zones | grep docker && firewall-cmd --zone=docker --list-all"));
+
+    script_retry("podman pull $test_img", retry => 3, delay => 60, timeout => 180);
+
+    # Start nginx in podman and validate initial ingress connectivity
+    assert_script_run("podman run -d --name nginx --rm $test_img");
+    my $nginx_podman_ip = script_output("podman container inspect nginx -f '{{ .NetworkSettings.IPAddress }}'");
+    assert_script_run("curl $nginx_podman_ip -svo/dev/null");
+
+    # Validate initial podman egress connectivity
+    assert_script_run("podman run --rm $test_img $test_cmd");
+
+    # Start docker and validate podman connectivity
+    systemctl("start docker");
+    systemctl("is-active docker");
+    systemctl("is-active firewalld");
+    if (script_run("curl $nginx_podman_ip -svo/dev/null") != 0) { bsc_1213811(); }
+    if (script_run("podman run --rm $test_img $test_cmd") != 0) { bsc_1213811(); }
+
+    # Prepare for docker validation
+    systemctl("stop firewalld");
+    systemctl("restart docker");
+    systemctl("is-active firewalld", expect_false => 1);
+    systemctl("is-active docker");
+
+    script_retry("docker pull $test_img", retry => 3, delay => 60, timeout => 180);
+
+    # Start nginx in docker and validate initial ingress connectivity
+    assert_script_run("docker run -d --name nginx --rm $test_img");
+    my $nginx_docker_ip = script_output("docker container inspect nginx -f '{{ .NetworkSettings.IPAddress }}'");
+    assert_script_run("curl $nginx_docker_ip -svo/dev/null");
+
+    # Validate initial docker egress connectivity
+    assert_script_run("docker run --rm $test_img $test_cmd");
+
+    # Start firewalld and validate docker connectivity
+    systemctl("restart firewalld");
+    systemctl("is-active firewalld");
+    if (script_run("curl $nginx_docker_ip -svo/dev/null") != 0) { bsc_1214080(); }
+    if (script_run("docker run --rm $test_img $test_cmd") != 0) { bsc_1214080(); }
+
+    # Once more validate podman connectivity with firewalld restarted
+    if (script_run("curl $nginx_podman_ip -svo/dev/null") != 0) { bsc_1214080(); }
+    if (script_run("podman run --rm $test_img $test_cmd") != 0) { bsc_1214080(); }
+}
+
+sub bsc_1213811 {
+    record_soft_failure('bsc#1213811 - podman network unreachable after starting docker');
+}
+
+sub bsc_1214080 {
+    record_soft_failure('bsc#1214080 - docker network broken after firewalld restart, if firewalld is disabled');
+}
+
+sub post_run_hook {
+    my ($self, $args) = @_;
+    my $podman = $self->containers_factory('podman');
```

Comment:
Nitpicking: Perhaps it would make sense to refactor a `cleanup` subroutine to avoid duplication in `post_run_hook` and in `post_fail_hook`.

Reply:
You mean something like?:
```perl
sub cleanup {
    my ($self, $args) = @_;
    my $podman = $self->containers_factory('podman');
    $podman->cleanup_system_host();
    my $docker = $self->containers_factory('docker');
    $docker->cleanup_system_host();
}
```